src/wasm/wat-parser.cpp

```diff
@@ -1255,6 +1257,27 @@ struct ParseDefsCtx : InstrParserCtx<ParseDefsCtx> {
     assert(local < func->getNumLocals());
     return push(pos, builder.makeLocalGet(local, func->getLocalType(local)));
   }
+
+  Result<> makeLocalTee(Index pos, Index local) {
+    if (!func) {
+      return in.err(pos, "local.tee must be inside a function");
+    }
+    assert(local < func->getNumLocals());
```

Comment:
Since the local index had to be produced by `getLocalFromIdx` or `getLocalFromName`, and those functions already checked for errors, it should definitely be valid if we get here.